tasks/configure.yml

```diff
@@ -158,8 +159,51 @@
     enabled: yes
   when: postgresql_service_enabled
 
-- name: PostgreSQL | Restart PostgreSQL
+- name: PostgreSQL | Ensure PostgreSQL is running
+  service:
+    name: "{{ postgresql_service_name }}"
+    state: started
+
+- name: PostgreSQL | Reload PostgreSQL to implement parameter changes | v9.5 and later
+  service:
+    name: "{{ postgresql_service_name }}"
+    state: reloaded
+  when: (
+          (postgresql_configuration_pt1.changed or
+           postgresql_configuration_pt2.changed or
+           postgresql_configuration_pt3.changed) and
+          postgresql_version | version_compare('9.5','>=')
+        )
+
+- name: PostgreSQL | Check for parameter changes that require a restart | v9.5 and later
+  become: yes
+  become_user: "{{ postgresql_service_user }}"
+  shell: "{{ postgresql_bin_directory}}/psql postgres --username {{postgresql_admin_user}} --tuples-only --command='SELECT count(*) COUNT FROM pg_settings WHERE pending_restart IS true;'"
+  register: postgresql_param_change_restart_required
+  when: postgresql_version | version_compare('9.5','>=')
+  changed_when: postgresql_param_change_restart_required.rc == 0 and
+                postgresql_param_change_restart_required.stdout | replace (' ','') != "0"
+
+- name: PostgreSQL | Force a checkpoint before a restart
+  become: yes
+  become_user: "{{ postgresql_service_user }}"
+  shell: "{{ postgresql_bin_directory}}/psql postgres --username {{postgresql_admin_user}} --tuples-only --command='CHECKPOINT;'"
+  register: postgresql_checkpoint_before_restart
+  when: postgresql_cluster_startup_conf.changed or
+        ( postgresql_param_change_restart_required.changed and
+          postgresql_restart_on_config_change
+        ) or
+        (
+          (postgresql_configuration_pt1.changed or
+           postgresql_configuration_pt2.changed or
+           postgresql_configuration_pt3.changed) and
+          postgresql_version | version_compare('9.5','<')
```

Comment:
The `pg_settings(pending_restart` only works on PostgreSQL v9.5 and later, so for earlier versions use the old behaviour of doing a full restart on any config change.